src/pac.c

```diff
@@ -112,10 +112,24 @@ pac_may_have_muzzy(pac_t *pac) {
 	return pac_decay_ms_get(pac, extent_state_muzzy) != 0;
 }
 
+size_t pac_batched_alloc_retained_size(size_t size) {
+	if (size > SC_LARGE_MAXCLASS) {
+		return size;
+	}
+	size_t batched_size = sz_s2u_size_class_compute(size);
+	size_t capped_size = ((size + HUGEPAGE - 1) >> LG_HUGEPAGE)
```

Comment:
Maybe
```
next_hugepage_size = HUGEPAGE_CEILING(batched_size)
return batched_size > next_hugepage_size ? next_hugepage_size : batched_size
```

Reply:
Sure, I think you mean `next_hugepage_size = HUGEPAGE_CEILING(size)`?